eox_nelp/openedx_filters/xapi/filters.py

```diff
@@ -198,3 +199,45 @@ def run_filter(self, transformer, result):  # pylint: disable=arguments-differ,
         return {
             "result": result
         }
+
+
+class XApiContextFilter(PipelineStep):
+    """This filter is designed to update the context value for any event transformer.
+
+    How to set:
+        OPEN_EDX_FILTERS_CONFIG = {
+            "event_routing_backends.processors.xapi.transformer.xapi_transformer.get_context": {
+                "pipeline": ["eox_nelp.openedx_filters.xapi.filters.XApiContextFilter"],
+                "fail_silently": False,
+            },
+        }
+    """
+
+    def run_filter(self, transformer, result):  # pylint: disable=arguments-differ, unused-argument
+        """Modifies statement context for any event
+
+        Arguments:
+            transformer <XApiTransformer>: Transformer instance.
+            result <Context>: Context related to an event.
+
+        Returns:
+            Context: Modified context.
+        """
+        extra_context = getattr(settings, "XAPI_EXTRA_CONTEXT", {})
+
+        for key, value in extra_context.items():
+            if not hasattr(result, key):
```

Comment:
Is not possible to add a new attribute if the result object doesn't have the attribute.
So this filter only updates attributes that the result has.
This could be specified in the docstring, but maybe we could need to add extra attributes or definitions hard-coded.

Reply:
There is a text with that. But why you wanted it in that way?
https://github.com/eduNEXT/eox-nelp/blob/39a9cc86fbc25f29a015fb2e1ea06baa8316330a/eox_nelp/openedx_filters/tests/xapi/tests_filters.py#L456-L468

Reply:
Tincan library just supports specific fields, so this was implemented in this way to follow the tincan standard fields https://github.com/RusticiSoftware/TinCanPython/blob/3.x/tincan/context.py#L42